db.go

```diff
@@ -124,12 +126,34 @@ func NewDB(dsn string) (*DB, error) {
 	return newDB, nil
 }
 
+// Creates the reputation table (or modifies it to match the schema we want). Done in a few
+// steps here to support migration of the schema from older to newer versions.
 const createReputationTableSQL = `
 CREATE TABLE IF NOT EXISTS reputation (
 ip ip4r PRIMARY KEY NOT NULL,
 reputation int NOT NULL CHECK (reputation >= 0 AND reputation <= 100)
 );
 CREATE INDEX IF NOT EXISTS reputation_ip_idx ON reputation USING gist (ip);
+
+DO $$
+	BEGIN
+		ALTER TABLE reputation ADD COLUMN reviewed boolean DEFAULT false;
+	EXCEPTION
+		WHEN duplicate_column THEN -- ignore error
+	END;
+$$;
+
+CREATE OR REPLACE FUNCTION reviewed_reset() RETURNS TRIGGER AS $$
+	BEGIN
+		NEW.reviewed = false;
+		RETURN NEW;
+	END;
+$$ LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS check_reviewed ON reputation;
+CREATE TRIGGER check_reviewed BEFORE UPDATE ON reputation
+	FOR EACH ROW WHEN (NEW.reputation = 100)
+	EXECUTE PROCEDURE reviewed_reset();
```

Comment:
Any reason we'd want to do this in here instead of the [update/decay lambda](https://github.com/mozilla-services/tigerblood/blob/master/tools/decay/decay.py)?

Reply:
Yeah, initially was thinking of putting it there but it didn't seem great to make the consistency/testing dependent on the polling interval of the lambda function. I think we'd want to make sure this flag is always off if the reputation is reset regardless of if it's decayed by this function or set by a client via one of the API calls.

We could add it to this Lambda function and also modify each insert/update on the reputation table to force the value in the go code (e.g., if rep == 100 default reviewed = false prior to update) vs. set it in the trigger as well, if we want to avoid the trigger?

I have no strong preference either way on this though, but not sure on more logic in the external function.

Reply:
Ugh, to summarize it sounds like our options are to use a db trigger or

* update in the decay lambda, but incur latency clearing `reviewed` of up to time between decay lambda calls and be unable to test it (without running the lambda locally as part of tests)
  * clear it from updates and inserts that set reputation to 100 to not have the latency penalty in that case

That is worse in terms of testability and code complexity, so I'm OK with the DB trigger.